backend/docker.go

```diff
@@ -405,18 +405,23 @@ func (p *dockerProvider) Start(ctx gocontext.Context, startAttributes *StartAttr
 		},
 	}
 
-	cpuSets, err := p.checkoutCPUSets(ctx)
-	if err != nil {
-		logger.WithFields(logrus.Fields{
-			"err":            err,
-			"cpu_set_length": len(p.cpuSets),
-			"run_cpus":       p.runCPUs,
-		}).Error("couldn't checkout CPUSets")
-		return nil, err
-	}
+	useCPUSets := p.runCPUs != uint(0)
+	cpuSets := ""
+
+	if useCPUSets {
```

Comment:
@soulshake This guard condition ensures that the CPU set checkout never happens when using CPU shares, which also means that the CPU set size isn't checked.  Does that help explain behavior?

Reply:
:+1: